src/xoj-preview-extractor/CMakeLists.txt

```diff
@@ -20,6 +20,7 @@ target_link_libraries (xournal-thumbnailer
   ${ZLIB_LIBRARIES}
   ${Glib_LDFLAGS}
   ${ZIP_LDFLAGS}
+  ${GTK_LDFLAGS}
```

Comment:
You will have to clean up the ld flags for macOS.
Here is how it is done:
```cmake
#
# DO NOT INCLUDE LIBRARIES WITH pkg_check_modules AFTER THIS LINE!!!
#
if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
    # Fix linker flag handling for MacOS builds where frameworks are separated from their flag
    string(REPLACE "-framework;" "-framework " xournalpp_LDFLAGS "${xournalpp_LDFLAGS}")
endif()
```

You could change this to a macro like add_include_ldflags and call it for all the LDFLAGS linked to the target.

Reply:
You will have to do so for all LDFLAGS and you should overwrite the variable as linking against it will lead to a failure anyway. Having the original variables available is actually bad as it can lead to errors.

Reply:
First build error was because of the `-framework` argument handling. Second build failure was because I made a typo in `thumbnailer_GTK_LDFLAGS`. It builds now.